akd_local_auditor/src/storage/s3/test.rs

```diff
@@ -0,0 +1,368 @@
+// Copyright (c) Facebook, Inc. and its affiliates.
+//
+// This source code is licensed under both the MIT license found in the
+// LICENSE-MIT file in the root directory of this source tree and the Apache
+// License, Version 2.0 found in the LICENSE-APACHE file in the root directory
+// of this source tree.
+
+//! A note to readers:
+//!
+//! We are marking most of the following tests as `#[ignored]` so they don't run automatically
+//! with `cargo test`. These require a docker container running `minio` to be running which is
+//! run in our CI pipeline when changes to this crate are made, but not otherwise since they
+//! are relatively expensive integration test operations. See [s3.yml](.github/workflows/s3.yml)
+//! for more information
+
+use crate::{
+    storage::{AuditProofStorage, EpochSummary},
+    Digest, Hasher,
+};
+
+use super::*;
+use akd::directory::Directory;
+use akd::ecvrf::HardCodedAkdVRF;
+use akd::storage::memory::AsyncInMemoryDatabase;
+use akd::AkdLabel;
+use akd::AkdValue;
+use anyhow::Result;
+use aws_config::SdkConfig;
+use aws_sdk_s3::{
+    config::Builder as S3ConfigBuilder,
+    model::{Delete, ObjectIdentifier},
+    Client,
+};
+use aws_smithy_http::byte_stream::ByteStream;
+
+// These are constants that are matched in both this crate's `docker-compose.yml`
+// and the s3.yml workflow pipeline
+const ACCESS_KEY: &str = "test";
+const SECRET_KEY: &str = "someLongAccessKey";
+const TEST_REGION: &str = "us-east-2";
+const TEST_ENDPOINT: &str = "http://127.0.0.1:9000";
+
+macro_rules! bucket_name {
+    () => {{
+        fn f() {}
+        fn type_name_of<T>(_: T) -> &'static str {
+            std::any::type_name::<T>()
+        }
+        let name = type_name_of(f);
+        name[..name.len() - 3]
+            .trim_end_matches("::{{closure}}")
+            .split("::")
+            .last()
+            .unwrap()
+            .replace("_", "")
+    }};
+}
+
+struct AuditInformation {
+    proof: akd::proof_structs::AppendOnlyProof<Hasher>,
+    phash: Digest,
+    chash: Digest,
+}
+
+/// Generate N audit proofs from a new tree
+async fn generate_audit_proofs(n: usize) -> Result<Vec<AuditInformation>, akd::errors::AkdError> {
+    let db = AsyncInMemoryDatabase::new();
+    let vrf = HardCodedAkdVRF {};
```

Comment:
We should probably add an associated new fn to this type: https://rust-lang.github.io/api-guidelines/predictability.html#c-ctor.